packages/wp-now/src/download.ts

```diff
@@ -11,6 +11,45 @@ import getWpNowPath from './get-wp-now-path';
 import getWordpressVersionsPath from './get-wordpress-versions-path';
 import getSqlitePath from './get-sqlite-path';
 import getWpCliPath from './get-wp-cli-path';
+import { HttpProxyAgent, HttpsProxyAgent } from 'hpagent';
+
+function httpsGet(url: string, callback: Function) {
+	const isBehindHttpProxy =
+		(process.env.http_proxy && process.env.http_proxy !== '') ||
+		(process.env.HTTP_PROXY && process.env.HTTP_PROXY !== '');
+
+	const isBehindHttpsProxy =
+		(process.env.https_proxy && process.env.https_proxy !== '') ||
+		(process.env.HTTPS_PROXY && process.env.HTTPS_PROXY !== '');
+
+	let agent;
+
+	if (isBehindHttpProxy || isBehindHttpsProxy) {
+		const urlParts = new URL(url);
+
+		let Agent;
+
+		if (urlParts.protocol === 'https:') {
+			Agent = HttpsProxyAgent;
+		} else if (urlParts.protocol === 'http:') {
+			Agent = HttpProxyAgent;
+		}
+
+		let proxy;
+
+		if (isBehindHttpsProxy) {
+			proxy = process.env.https_proxy ?? process.env.HTTPS_PROXY;
+		} else if (isBehindHttpProxy) {
+			proxy = process.env.http_proxy ?? process.env.HTTP_PROXY;
+		}
+
+		if (Agent && proxy) {
+			agent = new Agent({ proxy });
+		}
+	}
+
+	https.get(url, { agent }, callback);
+}
```

Comment:
@eruizdechavez , what do you think about simplifying this function a bit?
I can see some redundant checks around `isBehindHttpsProxy` and `proxy`.

Here is a suggestion:

```suggestion
function httpsGet(url: string, callback: Function) {
	const proxy =
		process.env.https_proxy ||
		process.env.HTTPS_PROXY ||
		process.env.http_proxy ||
		process.env.HTTP_PROXY;
	let agent;

	if (proxy) {
		const urlParts = new URL(url);
		const Agent =
			urlParts.protocol === 'https:' ? HttpsProxyAgent : HttpProxyAgent;
		agent = new Agent({ proxy });
	}

	https.get(url, { agent }, callback);
}
```

Reply:
I like the idea of simplifying the function. Let me try a couple of things and commit the updated version.

Reply:
Done.